xoinvader/enemy_wave.py

```diff
@@ -0,0 +1,78 @@
+"""Module for creating and maintaining waves of incoming enemies."""
+
+class EnemyWave(object):
+    """Container for enemy wave behavior.
+
+    Intended to use as just container, but may be subclassed. It's useful in
+    order to add methods for creating concrete animations and spawning actual
+    enemies.
+
+    Spawned enemies are tracked automatically and properly removed when
+
+    :param int speed: relative speed of the wave. Means how fast time advances.
+    The faster time advances, the shorter the delays between enemy spawns.
+    :param dict events: map from relative time in event to function that spawns
+    the enemies.
+    :param bool running: if wave currently advances.
+    """
+
+    def __init__(self):
+        self._running = False
+        self._counter = 0
+        self._speed = 0
+        self._events = {}
+        self._event_timeouts = []
+
+    @property
+    def speed(self):
+        return self._speed
+
+    @speed.setter
+    def speed(self, value):
+        self._speed = value
+
+    @property
+    def running(self):
+        return self._running
+
+    def add_event(self, time, callback):
+        """Add spawning event to some point in time.
+
+        :param int time: point relative to wave start when to run `callback`
```

Comment:
Oh, ticks != calls, I think... you must rely on game clock, not on call counter. I need to think about this.